src/modules/navigator/navigation.h

```diff
@@ -187,12 +199,19 @@ struct mission_fence_point_s {
 	double lat;
 	double lon;
 	float alt;
-	uint16_t nav_cmd;				/**< navigation command (one of MAV_CMD_NAV_FENCE_*) */
+
 	union {
-		uint16_t vertex_count;			/**< number of vertices in this polygon */
+		union {
```

Comment:
Why is this a union?

Reply:
@dagar how about adding a add CCASSERT to ensure this stays correct?

Reply:
Because it's within another union with a 2 byte vertex_count and a 4 byte circule_radius. This is to include the padding after the 2 byte vertex_count.

Reply:
The inner union does not add to the padding, _padding0 is just an alias for vertex_count. It should be a struct to add padding.

The rest looks good.

Reply:
Yes you're right, but actually it's not needed an all. It's an (unnecessary) artifact from before I moved nav_cmd down.

Updated.